packages/app-store/stripepayment/api/portal.ts

```diff
@@ -1,23 +1,37 @@
 import type { NextApiRequest, NextApiResponse } from "next";
 
+import { WEBAPP_URL } from "@calcom/lib/constants";
+
 import { getStripeCustomerIdFromUserId } from "../lib/customer";
 import stripe from "../lib/server";
 
 export default async function handler(req: NextApiRequest, res: NextApiResponse) {
-  if (req.method === "POST" || req.method === "GET") {
-    const customerId = await getStripeCustomerIdFromUserId(req.session!.user.id);
+  if (req.method !== "POST" && req.method !== "GET")
+    return res.status(405).json({ message: "Method not allowed" });
+  const { referer } = req.headers;
+
+  if (!referer) return res.status(400).json({ message: "Missing referrer" });
+
+  if (!req.session?.user?.id) return res.status(401).json({ message: "Not authenticated" });
 
-    if (!customerId) {
-      res.status(500).json({ message: "Missing customer id" });
-      return;
-    }
+  // If accessing a user's portal
+  const customerId = await getStripeCustomerIdFromUserId(req.session.user.id);
+  if (!customerId) return res.status(400).json({ message: "Missing customer id" });
 
-    const return_url = `${process.env.NEXT_PUBLIC_WEBAPP_URL}/settings/billing`;
-    const stripeSession = await stripe.billingPortal.sessions.create({
-      customer: customerId,
-      return_url,
-    });
+  let return_url = `${WEBAPP_URL}/settings/billing`;
 
-    res.redirect(302, stripeSession.url);
+  // If accessing a team's portal if referrer has /settings/team/[:teamId]/billing
+  if (/settings\/teams\/\d+\/billing/g.test(referer)) {
+    // Grab the teamId by just matching /settings/teams/[:teamId]/billing and getting third item in array after split
+    const teamId = referer.match(/\/(settings.+)/g) || "";
```

Comment:
This is not splitting the team id, right? 🤔  So.... teamId here equals an array with one item being the full string of the path?  Regex is not my strong suit, but might be wise to double check this ✌️

Reply:
Added `returnTo` query instead 🙏